0_RootFS/llvm_common.jl

```diff
@@ -1,3 +1,23 @@
+### Instructions for adding a new version
+#
+# * add the version/commit SHA to `llvm_tags`.  Commit SHAs can be found
+#   by viewing: https://github.com/llvm/llvm-project/releases
+# * create the directory `0_RootFS/LLVMBootstrap@X`.  You can copy the
+#   `build_tarballs.jl` file from `0_RootFS/LLVMBootstrap@X-1` and change the
+#   version to build.  Typically, to reduce patch duplication, we want to use
+#   symlinks if possible for patches.  Unfortunately, symlinks don't work
+#   with LLVMBootstrap so you'll need to copy all of the previous patches:
+#
+#      cp ../LLVMBootstrap@X-1/bundled .
+#
+# * you only need to build the platform `x86_64-linux-musl`. To deploy the shard
+#   and automatically update your BinaryBuilderBase's `Artifacts.toml`, use the
+#   `--deploy` flag to the `build_tarballs.jl` script.  You can build & deploy
+#   by running:
+#
+#      julia build_tarballs.jl --debug --verbose --deploy x86_64-linux-musl
```

Comment:
I'm not sure about the deploy instructions yet. `--deploy=local` seems to want to upload shards and just `--deploy` errors almost immediately with:
```
[ Info: Building and deploying version 11.0.1+0 to JuliaBinaryWrappers/LLVMBootstrap_jll.jl
ERROR: LoadError: GitError(Code:ERROR, Class:Object, cannot create branch `main` with `38a0f72851c7c380182710d40261766e346bf55b`)
Stacktrace:
 [1] branch!(repo::LibGit2.GitRepo, branch_name::String, commit::String; track::String, force::Bool, set_head::Bool)
   @ LibGit2 ~/julia/1.6/usr/share/julia/stdlib/v1.6/LibGit2/src/LibGit2.jl:438
 [2] init_jll_package(name::String, code_dir::String, deploy_repo::String; gh_auth::GitHub.OAuth2, gh_username::String)
   @ BinaryBuilder ~/.julia/packages/BinaryBuilder/hv55Q/src/AutoBuild.jl:921
 [3] init_jll_package(name::String, code_dir::String, deploy_repo::String)
   @ BinaryBuilder ~/.julia/packages/BinaryBuilder/hv55Q/src/AutoBuild.jl:874
 [4] build_tarballs(ARGS::Any, src_name::Any, src_version::Any, sources::Any, script::Any, platforms::Any, products::Any, dependencies::Any; julia_compat::String, kwargs::Any)
   @ BinaryBuilder ~/.julia/packages/BinaryBuilder/hv55Q/src/AutoBuild.jl:221
 [5] top-level scope
   @ ~/julia/Yggdrasil/0_RootFS/LLVMBootstrap@11/build_tarballs.jl:9
in expression starting at /home/omus/julia/Yggdrasil/0_RootFS/LLVMBootstrap@11/build_tarballs.jl:9

caused by: GitError(Code:ERROR, Class:Reference, cannot force update branch 'main' as it is the current HEAD of the repository.)
Stacktrace:
 [1] macro expansion
   @ ~/julia/1.6/usr/share/julia/stdlib/v1.6/LibGit2/src/error.jl:110 [inlined]
 [2] create_branch(repo::LibGit2.GitRepo, bname::String, commit_obj::LibGit2.GitCommit; force::Bool)
   @ LibGit2 ~/julia/1.6/usr/share/julia/stdlib/v1.6/LibGit2/src/reference.jl:238
 [3] branch!(repo::LibGit2.GitRepo, branch_name::String, commit::String; track::String, force::Bool, set_head::Bool)
   @ LibGit2 ~/julia/1.6/usr/share/julia/stdlib/v1.6/LibGit2/src/LibGit2.jl:435
 [4] init_jll_package(name::String, code_dir::String, deploy_repo::String; gh_auth::GitHub.OAuth2, gh_username::String)
   @ BinaryBuilder ~/.julia/packages/BinaryBuilder/hv55Q/src/AutoBuild.jl:921
 [5] init_jll_package(name::String, code_dir::String, deploy_repo::String)
   @ BinaryBuilder ~/.julia/packages/BinaryBuilder/hv55Q/src/AutoBuild.jl:874
 [6] build_tarballs(ARGS::Any, src_name::Any, src_version::Any, sources::Any, script::Any, platforms::Any, products::Any, dependencies::Any; julia_compat::String, kwargs::Any)
   @ BinaryBuilder ~/.julia/packages/BinaryBuilder/hv55Q/src/AutoBuild.jl:221
 [7] top-level scope
   @ ~/julia/Yggdrasil/0_RootFS/LLVMBootstrap@11/build_tarballs.jl:9
```
I suspect Yggdrasil CI is supposed to do the deployment?

Reply:
Seems that `--deploy=local` doesn't have the error I mentioned above and does an deployment: https://github.com/JuliaPackaging/Yggdrasil/releases/tag/LLVMBootstrap-v11.0.1

Reply:
One other thing to note is I noticed that `7z` was still being used to create the archive so there's another code path for archiving occurring which I'll need to uncover

Reply:
Probably in [`0_RootFS/common.jl`](https://github.com/JuliaPackaging/Yggdrasil/blob/8e88fdca6d71851c0bceb96665b08f66bfdb94c6/0_RootFS/common.jl).  Compiler shards are treated in a very special way

Reply:
Found the call to `7z` in BinaryBuilder: https://github.com/JuliaPackaging/BinaryBuilder.jl/blob/master/src/AutoBuild.jl#L818

Reply:
`--deploy` issues have been resolved